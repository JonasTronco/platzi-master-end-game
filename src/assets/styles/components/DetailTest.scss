@import '../main.scss';

.test-detail {
  text-align: justify;
  
  &__title {
    @include flexStaCen;
    margin-bottom: $space_x6;

    &__icon {
      margin-right: $space_x2;
      fill: $secondary_color;
      width: $space_x9;
      height: $space_x9;
    }
  }

  &__data {
    @include card;
    padding: $space_x6;

    &__option {
      fill: $secondary_color;
      width: $space_x6;
      height: $space_x6;
      margin-left: auto;

      &:hover {
        filter: brightness(95%);
        @include card;
      }
    }
    
    &__container--principal {
      @include flexStaSta;
      margin-bottom: $space_x6;
    }

    &__container {
      @include flexStaCen;
      margin: $space_x6 0px;

      div {
        margin: 0px;

        &:last-child {
          margin-left: auto;
          margin-right: auto;
        }
      }
    }

    &__element {
      @include colStaSta;
      margin: $space_x6 0px;
    }
  }
}