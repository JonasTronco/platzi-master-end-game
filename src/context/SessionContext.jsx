import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
