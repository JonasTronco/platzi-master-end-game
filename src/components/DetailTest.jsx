import PropTypes from 'prop-types';
import { useSession } from '../context/SessionContext';
import useRequest from '../hooks/useRequest';
import TestIcon from '../assets/icons/exam.svg';
import TestInfo from './TestInfo';
import '../assets/styles/components/DetailTest.scss';

function DetailTest({ testID }) {
  const { session } = useSession();
  const { response, loading, error } = useRequest(
    session.token,
    `${process.env.NEXT_PUBLIC_API_URL}/orders/${testID}`,
    1,
  );

  if (loading) {
    return <div className="loader" />;
  }

  if (error) {
    return <h3>{error.message}</h3>;
  }

  return (
    <main className="test-detail">
      {response && response.data ? (
        <>
          <section className="test-detail__title">
            <TestIcon className="test-detail__title__icon" />
            <div className="test-detail__title__text">
              <h2 className="test-detail__title__text__name">
                {response.data.name}
              </h2>
              <strong className="test-detail__title__text__shortName">
                {response.data.shortName}
              </strong>
            </div>
          </section>
          <TestInfo examName={response.data.shortName} />
        </>
      ) : (
        ''
      )}
    </main>
  );
}

DetailTest.propTypes = {
  testID: PropTypes.string.isRequired,
};

export default DetailTest;
