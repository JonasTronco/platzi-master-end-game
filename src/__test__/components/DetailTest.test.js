import DetailTest from '../../components/DetailTest';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('DetailTest Component', () => {
  const obj = {
    session: {
      token: '',
      user: {
        id: '',
        username: '',
        typeOfUser: '',
        isActive: true,
        imageURL: '',
        firstName: '',
        lastName: '',
        defaultPath: ''
      }
    }
  };
  test('Should render DetailTest component', () => {
    const component = create(
      <SessionProvider value={obj}>
        <DetailTest userID={''} testID={''} />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<DetailTest />', () => {
    const component = mount(
      <SessionProvider value={obj}>
        <DetailTest userID={''} testID={''} />
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
