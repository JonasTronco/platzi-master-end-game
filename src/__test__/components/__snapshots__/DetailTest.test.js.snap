// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailTest Component Should render DetailTest component 1`] = `
<main
  className="test-detail"
>
  
</main>
`;
